packages/lab/src/carousel/CarouselReducer.ts

```diff
@@ -0,0 +1,94 @@
+import type { Dispatch } from "react";
+
+import type { CarouselSlideId, CarouselSlideMeta } from "./CarouselSlide";
+
+export interface CarouselReducerState {
+  slides: Map<CarouselSlideId, CarouselSlideMeta>;
+  activeSlideIndex: number;
+  visibleSlides: number;
+  focusedSlideIndex: number;
+  carouselId?: string;
+}
+export type CarouselReducerAction =
+  | { type: "register"; payload: [CarouselSlideId, CarouselSlideMeta] }
+  | { type: "unregister"; payload: CarouselSlideId }
+  | { type: "updateSlideCount"; payload: number }
+  | { type: "move"; payload: CarouselSlideId }
```

Comment:
One for the radar, might need to update Stepper reducer which may still have goto. APIs should re-use naming and be consistent.